pyccel/parser/semantic.py

```diff
@@ -665,26 +706,19 @@ class instance persists after a call to a class method, and the arguments of a
         """
         assert not isinstance(exceptions, Variable)
         for i in self._allocs[-1]:
-            if isinstance(i, DottedVariable):
-                if isinstance(i.lhs.class_type, CustomDataType) and self._current_function != '__del__':
-                    continue
```

Comment:
What happened here?

Reply:
The coverage flagged it as unused and I couldn't find an example that triggers it. I don't know if that is because it was always unused or if my changes had an effect. It is possible that the allocated objects are now correctly associated with the scope of their class so there are no `DottedVariable`s in scopes anymore